t/ui/21-admin-needles.t

```diff
@@ -28,35 +28,44 @@ use Test::Mojo;
 use Test::Warnings ':all';
 use OpenQA::Test::Case;
 use Time::HiRes qw(sleep);
+use t::ui::PhantomTest;
 
 my $test_case = OpenQA::Test::Case->new;
 $test_case->init_data;
 
-# locations for dummy needle files
-my $needle_dir       = 't/data/openqa/share/tests/opensuse/needles/';
-my $needle_json_file = $needle_dir . 'inst-timezone-text.json';
-my $needle_png_file  = $needle_dir . 'inst-timezone-text.png';
-
 # ensure needle dir can be listed (might not be the case because previous test run failed)
+my $needle_dir = 't/data/openqa/share/tests/opensuse/needles/';
 chmod(0755, $needle_dir);
 
-# create dummy files for needle
-open(FNAME, '>', $needle_json_file);
-print FNAME 'go away later';
-close(FNAME);
-open(FNAME, '>', $needle_png_file);
-print FNAME 'go away later';
-close(FNAME);
-
-use t::ui::PhantomTest;
-
-my $driver = call_phantom();
+sub schema_hook {
+    my $schema  = OpenQA::Test::Database->new->create;
+    my $needles = $schema->resultset('Needles');
+    $needles->create(
+        {
+            dir_id                 => 1,
+            filename               => 'never-matched.json',
+            first_seen_module_id   => 3,
+            last_seen_module_id    => 10,
+            last_matched_module_id => undef,
+            file_present           => 1,
+        });
+}
+my $driver = call_phantom(\&schema_hook);
 unless ($driver) {
     plan skip_all => $t::ui::PhantomTest::phantommissing;
     exit(0);
 }
 
-is(-f $needle_json_file, 1, "file is created");
+my @needle_files = qw(inst-timezone-text.json inst-timezone-text.png never-matched.json never-matched.png);
+subtest 'create dummy files for needles' => sub {
+    for my $file (@needle_files) {
+        my $file_path = $needle_dir . $file;
+        #open(FNAME, '>', $file_path);
+        ok(open(FNAME, '>', $file_path), $file . ' is created');
+        print FNAME 'go away later';
+        close(FNAME);
```

Comment:
Symbolic references like `FNAME` are not really used anymore in modern Perl. Instead we just use lexical variables, like `open(my $fname, '>', $file_path)`. This also has the advantage that the handle is automatically closed once the variable goes out of scope.